content/blog/asyncapi_codegen_scst.md

```diff
@@ -0,0 +1,186 @@
+
+```java
+@SpringBootApplication
+public class Application {
+   private static final Logger logger = LoggerFactory.getLogger(Application.class);
+   public static void main(String[] args) {
+      SpringApplication.run(Application.class);
+   }
+
+   @Bean
+   public Consumer<RideReceipt> acmeRideshareBillingReceiptCreated001Consumer() {
+      // Add business logic here.
+      return null;
+   }
+}
+```
+
+You might say: “Marc, that’s great, but how the heck is that function actually binding to the messaging channels!?” This is where the `application.yml` file comes into play.
+
+The generated `application.yml` file defines several things as specified in the AsyncAPI document or from the parameters passed into the generator. First, it defines the list of functions it wants Spring Cloud Stream aware of under `spring.cloud.stream.function.definition`. Second, it tells Spring Cloud Stream which channels to bind those functions to under `spring.cloud.streams.bindings`. Lastly, it contains connection information to the messaging system. The connection info is specified by different parameters depending on the binder you choose but, in this case, it’s defined under `solace.java`.
+
+```yaml
+spring: 
+  cloud: 
+    stream: 
+      function: 
+        definition: acmeRideshareBillingReceiptCreated001Consumer 
+    bindings: 
+      acmeRideshareBillingReceiptCreated001Consumer-in-0:
+        destination: acme/rideshare/billing/receipt/created/0.0.1
+
+solace: 
+  java: 
+    host: 'localhost:55555' 
+    msgVpn: default 
+    clientUsername: default 
+    clientPassword: default
+
+logging: 
+  level: 
+    root: info 
+    org: 
+      springframework: info
+```
+
+Note that all of this was done for the developer so they didn’t have to track down which SCSt parameters needed to be set, map the functions to the bindings, etc. They just have to add their business logic in place of the project and hit run! In this case since it’s a Spring Boot project you can “run as a Spring Boot app” in your IDE or even run from the command line using `mvn spring-boot:run`.
+
+# **Helpful Parameters and Specification Extensions for Creating Microservices Using the AsyncAPI Spring Cloud Stream Template** 
+
+As I mentioned, there are a lot of complexities when it comes to generating useful application code from a microservice. Because of these complexities, I thought I’d call out some tips, tricks, and painpoints of using the AsyncAPI Spring Cloud Stream template.
+
+There are a bunch of different parameters and specification extensions that you should consider when generating your code. All of them can be found [here](https://github.com/asyncapi/java-spring-cloud-stream-template#configuration-options), but I’ll go over a few of the parameters that I use quite often:
+
+- **The `binder` parameter** allows you to specify the Spring Cloud Stream binder that you’d like to use. Currently the generator supports `kafka`, `rabbit` and `solace`.
```

Comment:
again to much bold, in UI it doesn't look good, you want to drag attention but if bold is overused the attention is lost, just leave binder in the code marks like you have `binder` and it will be fine, pink will drag enough attention :)

Reply:
fixed in next commit